### DISCUSSION TITLE
Automoderation feature similar to discord

### DISCUSSION THREAD

**Discussion by Andre601** [stoatchat/discussions#666] [Feature Suggestions]
Given the [recent spam attacks on Revolt](https://wiki.revolt.chat/en/transparency/incidents/2024-02-08-user-bot-spam), the need for an auto-mod feature similar to that of Discord gets ever more present, which is why I'm proposing such a feature to be added to Revolt.

The auto-mod feature should perform actions **before** the message is even sent, with bots having the option to include rule-sets and actions of their own.

The following actions should be considered:

- Report message to one or multiple channels
- Block message from sending
- Mute user (temporarely)

In addition can I think of the following auto-mod rules to exist by default, that could be enabled within a server's settings:

- Anti-mass-mention (Configurable amount of mentions to perform actions on)
- Anti-invite (With whitelist allowing sets of invite codes, or alternatively server IDs)

I'm willing to draft an RFC in the RFC repository for this feature to be considered.

**Comment by Rexogamer** [comment]
demonic

**Comment by YukitoNakamura** [comment]
Automoderation would depend on moderation bots, which is the same system most Discord servers use.

**Reply by IAmTomahawkx** [reply]
discord's built in automod does not rely on bots.

**Reply by YukitoNakamura** [reply]
Oops, my bad then. Even though I never managed to learn the automoderation on Discord and always relied on bots for it. Thank you for the info.

**Reply by ghost** [reply]
Revolt users cloud flare for spam and ddos protection however we need something built into the app I agree it should not be like discord automod since that incorrectly flags content we need more verification features and possibly our own ddos protection service along with the built in cloud flare cdn.
